### DISCUSSION TITLE
failed to create registration token for github enterprise server

### DISCUSSION THREAD

**Discussion by karnatamp** [actions/actions-runner-controller#2029] [Questions]
I tried to deploy ARC in google kubernetes engine using github enterprise server PAT authentication it is showing failed to create registration token. but when i use normal github( personal github account instead organization) it is working. so i want to what would be the problem how to resolve this issue?

please let me know if anyone have solution for this situation, thank you.

**Comment by mspasoje** [comment]
Not sure if it applies for your scenario but for us this simple runner deployment setup worked (enterprise v3.6) for repository or organization runners:
1. Make sure you separate work of multiple ARCs in case you have them in the same kuberentes cluster. For us this meant to make sure each ARC monitors single namespace. Having multiple ARCs in the same cluster monitoring overlapping namespaces (or 'all namespaces') with one maintaining runners on 'normal github' and one on Enterprise Github did not work. More on this [here](https://github.com/actions/actions-runner-controller/blob/master/docs/deploying-arc-runners.md). This is facilitated by (see below complete values.yaml):
`scope:`
`  watchNamespace: enterprise-action-runners`
`  singleNamespace: true`
2. We used helm and installed ARC into it's own namespace, command like this: 
`helm upgrade --install --namespace actions-runner-system-namespace --create-namespace -f values.yaml --wait actions-runner-controller-dev actions-runner-controller/actions-runner-controller`

values.yaml:

`githubURL: enterprise_github_url/api/v3`
`githubUploadURL: enterprise_github_url/api/uploads`
`runnerGithubURL: enterprise_github_url`
`scope:`
`  watchNamespace: enterprise-action-runners`
`  singleNamespace: true`
`authSecret:`
`  create: true`
`  github_token: your_PAT_here`

For us enterprise_github_url is: https://github.enterprise_name.com with of course enterprise_name replaced with the real name.

To deploy runners as RunnerDeployment, appropriate yaml looks somewhat like this:
`apiVersion: actions.summerwind.dev/v1alpha1`
`kind: RunnerDeployment`
`metadata:`
`  name: some_name`
`  namespace: enterprise-action-runners`
`spec:`
`  replicas: 2`
`  template:`
`    spec:`
`      repository: org_name/repo_name`
`      organization: org_name`
`      image: summerwind/actions-runner:ubuntu-20.04`

namespace should match the namespace monitored by ARC (above), org_name and repo_name as appropriate. (!)Make sure to comment out either repository or organization as runners can only monitor one thing at a time. Don't forget to create namespace and apply runnerdeployment.yaml in that namespace:
`kubectl create namespace enterprise-action-runners`
`kubectl apply -f runnerdeployment.yaml -n enterprise-action-runners`
Also, make sure your PAT is configured as mentioned [here](https://github.com/actions/actions-runner-controller/blob/master/docs/authenticating-to-the-github-api.md).

Probably worth mentioning is that GitHub app authorized runners did not work for us for the same setup. In other words, same configuration with replacing github_token with github_app_id, github_app_installation_id and github_app_private_key could not start runners.
